gnss_lib_py/utils/sv_models.py

```diff
@@ -0,0 +1,611 @@
+"""Model GNSS SV states (positions and velocities).
+
+Functions to calculate GNSS SV positions and velocities for a given time.
+"""
+
+__authors__ = "Ashwin Kanhere, Bradley Collicott"
+__date__ = "17 Jan, 2023"
+
+import warnings
+import numpy as np
+
+import gnss_lib_py.utils.constants as consts
+from gnss_lib_py.utils.coordinates import ecef_to_el_az
+from gnss_lib_py.parsers.navdata import NavData
+from gnss_lib_py.parsers.ephemeris import EphemerisManager
+from gnss_lib_py.utils.time_conversions import gps_millis_to_tow, gps_millis_to_datetime
+
+
+def svs_from_el_az(elaz_deg):
+    """Generate NED satellite positions at given elevation and azimuth.
+
+    Given elevation and azimuth angles are with respect to the receiver.
+    Generated satellites are in the NED frame of reference with the receiver
+    position as the origin.
+
+    Parameters
+    ----------
+    elaz_deg : np.ndarray
+        Nx2 array of elevation and azimuth angles [degrees]
+
+    Returns
+    -------
+    svs_ned : np.ndarray
+        Nx3 satellite NED positions, simulated at a distance of 20,200 km
+    """
+    assert np.shape(elaz_deg)[0] == 2, "elaz_deg should be a 2xN array"
+    el_deg = np.deg2rad(elaz_deg[0, :])
+    az_deg = np.deg2rad(elaz_deg[1, :])
+    unit_vect = np.zeros([3, np.shape(elaz_deg)[1]])
+    unit_vect[0, :] = np.sin(az_deg)*np.cos(el_deg)
+    unit_vect[1, :] = np.cos(az_deg)*np.cos(el_deg)
+    unit_vect[2, :] = np.sin(el_deg)
+    svs_ned = 20200000*unit_vect
+    return svs_ned
+
+
+def add_sv_states(measurements, ephemeris_path, constellations=['gps'], delta_t_dec = -2):
```

Comment:
Can we figure out how to do an `inplace` version? The question is what to do with the rows that don't match the provided constellation (e.g. SBAS, glonass right now). Remove them? Add in NaN for their respective sv_state?